### PR TITLE
#82 - GitHub Actions

### PR DESCRIPTION
**Pull request by Jeckerson**
Issue - #82

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+  run:
+    runs-on: ubuntu-latest
+    services:
+      mysql:
+        image: mysql:5.7
+        env:
+          MYSQL_ROOT_PASSWORD: root
+    steps:
+      env:
+        DB_HOST: 127.0.0.1
```
Comment (sergeyklay) [inline .github/workflows/ci.yml]:
```suggestion
        DB_HOST: '127.0.0.1'
```

Comment (sergeyklay) [reply .github/workflows/ci.yml]: You have cast to string such numbers

Comment (Jeckerson) [reply .github/workflows/ci.yml]: 4e4197b

File: tests/functional/Forms/ProductTypesFormTest.php [hunk 2]
```diff
@@ -30,6 +30,8 @@ public function inputDataProvider(): array
      */
     public function testValidation(array $data, bool $expected): void
     {
+        Di::reset();
```
Comment (sergeyklay) [inline tests/functional/Forms/ProductTypesFormTest.php]: Should we move this to setUp ?

File: tests/functional/Models/ContactTest.php [hunk 3]
```diff
@@ -13,6 +13,8 @@ final class ContactTest extends Unit
 {
     public function testBeforeCreate(): void
     {
+        Di::reset();
```
Comment (sergeyklay) [inline tests/functional/Models/ContactTest.php]: The same

File: .github/workflows/ci.yml [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+          mysql -uroot -h127.0.0.1 -proot -e 'CREATE DATABASE IF NOT EXISTS `invo`;'
+
+      - name: Copy .env file
+        run: cp .env.example .env
+
+      - name: Add Phalcon package repository
+        run: curl -s https://packagecloud.io/install/repositories/phalcon/mainline/script.deb.sh | sudo bash
+
+      - name: Install PSR and Xdebug PECL Packages
+        run: sudo apt-get install php-pear php-xdebug && sudo pecl install --force psr
```
Comment (sergeyklay) [inline .github/workflows/ci.yml]: There is an action with phalcon support. Why do we need all this?

File: .github/workflows/ci.yml [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+
+      - name: Init Database
+        run: |
+          mysql -uroot -h127.0.0.1 -proot -e 'CREATE DATABASE IF NOT EXISTS `invo`;'
+
+      - name: Copy .env file
+        run: cp .env.example .env
+
+      - name: Add Phalcon package repository
+        run: curl -s https://packagecloud.io/install/repositories/phalcon/mainline/script.deb.sh | sudo bash
```
Comment (sergeyklay) [inline .github/workflows/ci.yml]: Do we really need install Phalcon by hand?

File: .github/workflows/ci.yml [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+        run: composer install --prefer-dist --no-progress --no-suggest
+
+      - name: Install PHPStan
+        run: composer require --prefer-dist --no-progress --no-suggest --dev phpstan/phpstan
+
+      - name: Run Psalm
+        run: vendor/bin/psalm --show-info=false
+
+      - name: Run PHPStan
+        run: vendor/bin/phpstan analyse -l 5 src
```
Comment (sergeyklay) [inline .github/workflows/ci.yml]: Could you move all stat. analyse the separated workflow?

File: .github/workflows/ci.yml (hunk 5 above)
Comment (Jeckerson) [reply .github/workflows/ci.yml]: Copy/paste from migrations .yml. Need to change in both.

File: tests/functional/Forms/ProductTypesFormTest.php (hunk 2 above)
Comment (Jeckerson) [reply tests/functional/Forms/ProductTypesFormTest.php]: 84d79ab

File: tests/functional/Models/ContactTest.php (hunk 3 above)
Comment (Jeckerson) [reply tests/functional/Models/ContactTest.php]: 84d79ab

File: .github/workflows/ci.yml [hunk 7]
```diff
@@ -0,0 +1,57 @@
...
+
+      - name: Init Database
+        run: |
+          mysql -uroot -h127.0.0.1 -proot -e 'CREATE DATABASE IF NOT EXISTS `invo`;'
+
+      - name: Copy .env file
+        run: cp .env.example .env
+
+      - name: Setup PHP
+          uses: shivammathur/setup-php@v1
```
Comment (sergeyklay) [inline .github/workflows/ci.yml]: wrong indent

File: .github/workflows/ci.yml [hunk 8]
```diff
@@ -0,0 +1,57 @@
...
+          extensions: mbstring, intl, json, phalcon4
+
+      - name: Validate composer.json and composer.lock
+        run: composer validate
+
+      - name: Install Composer dependencies
+        run: composer install --prefer-dist --no-progress --no-suggest
+
+      - name: Install PHPStan
+        run: composer require --prefer-dist --no-progress --no-suggest --dev phpstan/phpstan
```
Comment (sergeyklay) [inline .github/workflows/ci.yml]: AFAIK, this command will force call `install` so you don't need previous call

File: .github/workflows/ci.yml [hunk 9]
```diff
@@ -0,0 +1,57 @@
...
+
+      - name: Copy .env file
+        run: cp .env.example .env
+
+      - name: Setup PHP
+        uses: shivammathur/setup-php@v1
+        with:
+          php-version: '7.2'
+          tools: pecl
+          extensions: mbstring, intl, json, phalcon4
```
Comment (sergeyklay) [inline .github/workflows/ci.yml]: Add `coverage:` tool here

File: .github/workflows/ci.yml [hunk 10]
```diff
@@ -0,0 +1,57 @@
...
+        run: composer validate
+
+      - name: Install Composer dependencies
+        run: composer install --prefer-dist --no-progress --no-suggest
+
+      - name: Install PHPStan
+        run: composer require --prefer-dist --no-progress --no-suggest --dev phpstan/phpstan
+
+      - name: Run Psalm
+        run: vendor/bin/psalm --show-info=false
```
Comment (sergeyklay) [inline .github/workflows/ci.yml]:
```suggestion
        if: any()
        run: vendor/bin/psalm --show-info=false
```

File: .github/workflows/ci.yml [hunk 11]
```diff
@@ -0,0 +1,57 @@
...
+
+      - name: Run PHPStan
+        run: vendor/bin/phpstan analyse -l 5 src
```
Comment (sergeyklay) [inline .github/workflows/ci.yml]:
```suggestion
        if: any()
        run: vendor/bin/phpstan analyse -l 5 src
```

File: README.md [hunk 12]
```diff
@@ -56,12 +56,3 @@ Support us with a monthly donation and help us continue our activities. [[Become
 ## License
 
 Invo is open-sourced software licensed under the [New BSD License][8]. © Phalcon Framework Team and contributors
-
-[1]: https://phalcon.io/
```
Comment (sergeyklay) [inline README.md]: There are links like `See [CONTRIBUTING.md][7]`

Comment (Jeckerson) [reply README.md]: Do we need this links here?

Comment (sergeyklay) [reply README.md]: all these or just this? You just deleted all links from the document

Comment (Jeckerson) [reply README.md]: Reverted

**Pull request merged**